proposals/lip-add_weights_to_lisk-bft_consensus_protocol.md

```diff
@@ -0,0 +1,314 @@
+```
+LIP: <LIP number>
+Title: Add weights to Lisk-BFT consensus protocol
+Author: Jan Hackfeld <jan.hackfeld@lightcurve.io>
+Discussions-To: https://research.lisk.com/t/add-weights-to-lisk-bft-consensus-protocol/289
+Type: Standards Track
+Created: <YYYY-MM-DD>
+Updated: <YYYY-MM-DD>
+Requires: Introduce BFT module LIP
+```
+
+## Abstract
+
+This LIP defines how to generalize the Lisk-BFT consensus protocol to allow for different BFT weights of the participating validators, which may also change over time.
+The BFT weight is the weight attributed to the prevotes and precommits cast by a validator and therefore determines to what extent the validator contributes to finalizing blocks.
+We specify this generalized computation of consensus votes and associated properties using the new state structure defined in the BFT module.
+
+## Copyright
+
+This LIP is licensed under the [Creative Commons Zero 1.0 Universal](https://creativecommons.org/publicdomain/zero/1.0/).
+
+## Motivation
+
+The Lisk-BFT consensus protocol introduced in [LIP 0014][lisk-bft-lip] was specified for the Lisk mainchain and sidechains that use Delegated Proof-of-Stake with fixed configuration parameters that only change in rare cases as part of a hard fork.
+In these blockchains, there is a fixed number of active delegates, i.e., those delegates participating in finalizing blocks, and all of them have equal weight.
+One exception to that is the bootstrap period introduced as part of the new genesis block format in [LIP 0034][lip-34].
+During the bootstrap period none of the bootstrap delegates contributes to finalizing blocks, so that blocks can only be finalized after the bootstrap period is over.
+
+For more flexibility and to also support other validator selection mechanisms, we want to allow for different BFT weights of the active validators, i.e., different weights attributed to the prevotes and precommits cast by a validator.
+These weights and the threshold for finalizing a block should also not be constant, but should be allowed to change over time. This is, in particular, required in conjunction with the Proof-of-Authority validator selection mechanism introduced in the [PoA][module] where validators can be added and removed or their weights changed with a simple transaction and without a hard fork.
```

Comment:
This link to the PoA module seems broken.

Reply:
I fixed the link.